File: versioned_docs/version-1.12.x/examples/cpp/src/data_management_delete_query.cc
#include <reduct/client.h>
#include <cassert>
#include <chrono>

using reduct::IBucket;
using reduct::IClient;
using reduct::Error;
using std::chrono_literals::operator ""s;

int main() {
    // Create a client instance, then get or create a bucket
    auto client = IClient::Build("http://127.0.0.1:8383", {.api_token="my-token"});
    auto [bucket, create_err] = client->GetOrCreateBucket("my-bucket");
    assert(create_err == Error::kOk);

    //  Send some records to the "cpp-example"
    IBucket::Time ts = IBucket::Time::clock::now();
    auto err = bucket->Write("cpp-example", ts, [](auto rec) {
        rec->WriteAll("Some binary data");
    });
    assert(err == Error::kOk);

    err = bucket->Write("cpp-example", ts + 1s, [](auto rec) {
        rec->WriteAll("Some binary data");
    });
    assert(err == Error::kOk);

    // Delete all records withing a time range
    auto [removed_records, query_err] = bucket->RemoveQuery("cpp-example", ts, ts + 2s, {});
    assert(query_err == Error::kOk);
    assert(removed_records == 2);

    // You can also delete all records with a specific label
    bucket->RemoveQuery("cpp-example", {}, {},{
        .include = {{"key1", "value1"}}
    });

    // Or each N-th record
    bucket->RemoveQuery("cpp-example", {}, {},{
        .each_n = 2
    });
}

File: versioned_docs/version-1.12.x/examples/js/src/data_management_delete_query.mjs
import { Client } from "reduct-js";
import assert from "node:assert";

// Create a client instance, then get or create a bucket
const client = new Client("http://127.0.0.1:8383", { apiToken: "my-token" });
const bucket = await client.getOrCreateBucket("bucket");

// Send some records to the "entry-1" entry
const timestamp = BigInt(Date.now()) * 1000n;
let record = await bucket.beginWrite("entry-1", timestamp);
await record.write("Some binary data");

record = await bucket.beginWrite("entry-1", timestamp + 1000_000n);
await record.write("Some more binary data");

// Delete all records withing a time range
const removedRecords = await bucket.removeQuery(
  "entry-1",
  timestamp,
  timestamp + 2000_000n,
);
assert(removedRecords === 2);

// You can also delete all records with a specific label
await bucket.removeQuery("entry-1", undefined, undefined, {
  include: { label1: "value1" },
});

// Or each N-th record
await bucket.removeQuery("entry-1", undefined, undefined, { eachN: 2 });
